### PR TITLE
⭐️ Adds platformDetails to aws.ec2.instance resources

### PR DESCRIPTION
**Pull request by scottford-io**
This PR adds the `platformDetails` field for EC2 instances to the `aws.ec2.instance` resource to query query/filter on specific platforms (Linux/Unix/Windows/etc). 

image

image

### CODE REVIEW

**Review by vjeffrey — APPROVED** [review]

File: resources/packs/aws/aws_ec2.go [hunk 1]
```diff
@@ -716,6 +716,7 @@ func (s *mqlAwsEc2) gatherInstanceInfo(instances []types.Reservation, imdsvVersi
 			}
 
 			args := []interface{}{
+				"platform", core.ToString(instance.PlatformDetails),
```
Comment (chris-rock) [inline resources/packs/aws/aws_ec2.go]: are we sure we want to connect PlatformDetails to `platform`

**Comment by scottford-io** [comment]
@chris-rock should I update it to `platformDetails`?

File: resources/packs/aws/aws_ec2.go (hunk 1 above)
Comment (scottford-io) [reply resources/packs/aws/aws_ec2.go]: @chris-rock @vjeffrey I have updated field to be `platformDetails` to match the AWS Go SDK

```
 	// The platform details value for the instance. For more information, see AMI
	// billing information fields
	// (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/billing-info-fields.html)
	// in the Amazon EC2 User Guide.
	PlatformDetails *string
```

image

Comment (scottford-io) [reply resources/packs/aws/aws_ec2.go]: Please let me know if you would like other changes.

Comment (chris-rock) [reply resources/packs/aws/aws_ec2.go]: Do we have platform available in the API as well? Looks like in general it `platform` is easier to validate on

**Pull request merged**